src/lib/device_tree/src/lib.rs

```diff
@@ -9,11 +26,25 @@ use core::fmt;
 use model::{Driver, Result};
 use wrappers::SectionReader;
 
+/// Special value that every device tree in FDT format starts with.
 pub const MAGIC: u32 = 0xd00dfeed;
+// TODO: Explain why max depth is 16.
 pub const MAX_DEPTH: usize = 16;
-// Maximum length of a node or property name. This is 31 for the name, 32 for the unit name and 1
-// for the null terminator. An error is returned for longer names.
-pub const MAX_NAME_SIZE: usize = 31 + 32 + 1;
+/// Maximum length of a node or property name.
+///
+/// According to standard, the name has format `node-name@unit-address`:
+///
+/// * the `node-name` part is required to be between 1-31 characters (bytes)
+/// * `unit-address` is optional, but if present, should contain address in hex format.
+///   We are reserving 32 characters to encode 128 bit address.
+/// * We are reserving 1 byte for null terminator.
+///
+/// An error is returned for longer names.
+///
+/// TODO: Reserve 1 byte for "@" character.
+/// TODO: Do we need to encode 128 bit addresses, or would it be enough to just support
```

Comment:
Current the representation of the name is just a string, we don't try to parse the components (yet).

Reply:
Yes, in this comment I am just trying to reverse engineer why this constant is "31 + 1 + 32" and I wasn't sure where the "32" part came from. My best bet was that it was used to reserve a little more space for this "unit-address" component. From my understanding, it is supposed to be a hex number, so 32 bytes would allow us to store the 128 bit address. This TODO is to figure out if this 32 bytes was intentional to represent 128 bits, or was it used mostly because it nicely double the other 32 bytes, and the resulting 64 will probably align well in memory.

Reply:
Your analysis looks right. The number is basically arbitrary, but has to be "big enough". If I recall, the spec does not say anything about a maximum length. But, for practical reasons, we limit it.

As for the "31 + 1 + 32", it does look like it is intended to reserve space for different parts. Feel free to replace this with "64" if it will make the docs easier.